core/controllers/suggestion.py

```diff
@@ -50,6 +50,38 @@ def _require_valid_suggestion_and_target_types(target_type, suggestion_type):
             'Invalid suggestion_type: %s' % suggestion_type)
 
 
+def _render_suggestions(self, target_type, suggestions):
+    """Renders retrieved suggestions for a handler."""
+    target_ids = set([s.target_id for s in suggestions])
```

Comment:
Again, instead of calling self.render_json() probably it just constructs the overall dict needed and returns it? The controller should call render_json IMO.

Reply:
I created a parent class with a helper method as suggested above.